File: ted-ext/ted-scheduler/src/main/java/ted/driver/sys/_TedSchdDriverExt.java
package ted.driver.sys;


import ted.driver.TedDriver;
import ted.driver._TedSchdHck;

import javax.sql.DataSource;

/**
 * @author Augustus
 *         created on 2018.08.17
 *
 *  use some of internal functions.
 *
 *  For ted-scheduler only!
 *
 */
public class _TedSchdDriverExt {
	private final TedDriver tedDriver;
	private final TedDriverImpl tedDriverImpl;

	public _TedSchdDriverExt(TedDriver tedDriver) {
		this.tedDriver = tedDriver;
		this.tedDriverImpl = _TedSchdHck.getTedDriverImpl(tedDriver);
	}

	public String systemId() {
		return tedDriverImpl.getContext().config.systemId();
	}
	public String instanceId() {
		return tedDriverImpl.getContext().config.instanceId();
	}
	public Long primeTaskId() {
		return tedDriverImpl.getContext().prime.primeTaskId();
	}
	public DataSource dataSource() {
		return tedDriverImpl.dataSource;
	}
	public boolean isPrimeEnabled() {
		return tedDriverImpl.prime().isEnabled();
	}
}

File: ted-ext/ted-scheduler/src/main/java/ted/scheduler/impl/_hck/_TedSchdDriverExt.kt
package ted.driver.sys


import ted.driver.TedDriver
import ted.driver._TedSchdHck
import ted.driver.sys.SqlUtils.DbType
import javax.sql.DataSource

/**
 * @author Augustus
 * created on 2018.08.17
 *
 * use some of internal functions.
 *
 * For ted-scheduler only!
 */
class _TedSchdDriverExt(private val tedDriver: TedDriver) {
    private val tedDriverImpl: TedDriverImpl
    val isPrimeEnabled: Boolean
        get() = tedDriverImpl.prime()!!.isEnabled()

    init {
        this.tedDriverImpl = _TedSchdHck.getTedDriverImpl(tedDriver)

    }

    fun systemId(): String {
        return tedDriverImpl.context.config.systemId()
    }

    fun instanceId(): String {
        return tedDriverImpl.context.config.instanceId()
    }

    fun primeTaskId(): Long? {
        return tedDriverImpl.context.prime.primeTaskId()
    }

    fun dataSource(): DataSource {
        return tedDriverImpl.dataSource
    }

    fun dbType() : DbType {
        return tedDriverImpl.context.tedDao.dbType;
    }
}
